Composer/packages/lib/indexers/src/botIndexer.ts

```diff
@@ -38,7 +47,8 @@ const checkSkillSetting = (assets: BotAssets): Diagnostic[] => {
   dialogs.forEach((dialog) => {
     // used skill not existed in setting
     dialog.skills.forEach((skillId) => {
-      if (skill.findIndex(({ manifestUrl }) => manifestUrl === skillId) === -1) {
+      const endpointUrlCollection = map(skill, ({ endpointUrl }) => endpointUrl);
+      if (!endpointUrlCollection.includes(fetchFromSettings(skillId, assets.setting))) {
         diagnostics.push(
           new Diagnostic(`skill '${skillId}' is not existed in appsettings.json`, dialog.id, DiagnosticSeverity.Error)
```

Comment:
The grammar on this doesn't make sense. Now might be a good time to fix it while were making changes.
```suggestion
          new Diagnostic(`skill '${skillId}' does not exist in appsettings.json`, dialog.id, DiagnosticSeverity.Error)
```

Reply:
Any specific suggestions. How about `The skill ${skillName} does not exist in appsettings.json`?

Reply:
Yea that's perfect